financial-connections/src/main/java/com/stripe/android/financialconnections/features/consent/ui/ConsentLogoHeader.kt

```diff
@@ -139,30 +129,37 @@ private fun debugPreviewBitmaps(
 private fun loadBitmaps(
     logos: List<String>,
     bitmapLoadSize: Int,
-    isVisible: MutableState<Boolean>
-): SnapshotStateList<ImageBitmap?> {
+): List<ImageBitmap?> {
     val stripeImageLoader = LocalImageLoader.current
-    val loadedImages = remember { mutableStateListOf<ImageBitmap?>() }
-    // Load images
-    logos.forEachIndexed { index, logo ->
-        LaunchedEffect(logo) {
-            stripeImageLoader.load(
-                logo,
-                bitmapLoadSize,
-                bitmapLoadSize
-            ).onSuccess { result ->
-                while (loadedImages.size <= index) {
-                    loadedImages.add(null)
+    val loadedImages = remember { mutableStateListOf(*Array<ImageBitmap?>(logos.size) { null }) }
+    val placeholderBitmap = generatePlaceholderBitmap(bitmapLoadSize, colors.textCritical)
+
+    LaunchedEffect(logos) {
+        logos.forEachIndexed { index, logo ->
+            stripeImageLoader.load(logo, bitmapLoadSize, bitmapLoadSize).let { loadResult ->
+                loadResult.onSuccess { result ->
+                    loadedImages[index] = result?.asImageBitmap()
                 }
-                loadedImages[index] = result?.asImageBitmap()
-                // Set visibility to true when all images are loaded
-                if (loadedImages.size == logos.size && loadedImages.all { it != null }) {
-                    isVisible.value = true
+                loadResult.onFailure {
+                    Log.e("ConsentLogoHeader", "Failed to load image: $logo")
+                    // TODO handle image failure.
                 }
             }
         }
     }
-    return loadedImages
+
+    // TODO: figure how to trigger recomposition on bitmap changes, rather than relying on nullability.
+    // Initially, the SnapshotStateList is filled with nulls, so return a list with placeholders first.
+    // Once the images are loaded, they'll replace the nulls and trigger recomposition.
+    return loadedImages.map { it ?: placeholderBitmap }
+}
+
+private fun generatePlaceholderBitmap(bitmapLoadSize: Int, placeholderColor: Color): ImageBitmap {
```

Comment:
We should turn this into `rememberPlaceholderBitmap` to avoid this being called on every composition.